src/Pollers/SnmpPoller.php

```diff
@@ -144,14 +152,20 @@ private function updateStatusAfterException(SNMPException $exception):array
     private function pollDevices($chunks, $fileUniquePrefix, $counter)
     {
         $handle = fopen("/tmp/$fileUniquePrefix" . "_sonar_$counter","w");
+		//this allows a savvy user to be able to determine which threads are failing and can whittle down the hosts causing the problems
+		$output = fopen("/tmp/$fileUniquePrefix" . "_HOST_$counter","w");
+		fwrite($output, json_encode($chunks));
+		fclose($output);
```

Comment:
Maybe wrap this part in the debug check too?

Reply:
debug check added